### DISCUSSION TITLE
moving our GitHub repository to `jax-ml`

### DISCUSSION THREAD

**Discussion by froystig** [jax-ml/jax#23319] [Announcements]
JAX's GitHub repository is moving to the [`jax-ml` GitHub org](https://github.com/jax-ml/) in September 2024\.

Summary:

* The canonical JAX repository is relocating to [github.com/jax-ml/jax](http://github.com/jax-ml/jax).  
* URLs and forks of [github.com/google/jax](http://github.com/google/jax) will redirect to the new location.  
* This will improve our tooling options for continuous integration and testing.  
* This does not reflect any change in how JAX is developed (aside from the URL).

Coming soon, the JAX core repository will relocate from [github.com/google/jax](http://github.com/google/jax) to [github.com/jax-ml/jax](http://github.com/jax-ml/jax). GitHub takes care to help avoid any interruption due to the move – see the [GitHub docs on transferring repositories](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository). In particular, forks and HTTPS/SSH requests will automatically redirect to the new location, indefinitely.

The JAX core team has operated the `jax-ml` GitHub organization for several years. It houses a handful of repositories related to the JAX core—such as [`ml_dtypes`](https://github.com/jax-ml/ml\_dtypes) and [`jax-triton`](https://github.com/jax-ml/jax-triton)—plus a few more experimental projects. Meanwhile, The JAX core repository has been located within the broader [Google-wide GitHub org](http://github.com/google/) since its initial open-source launch in 2018, before we had `jax-ml`.

Our `jax-ml` GitHub org offers us useful features, especially ones that we can use to improve the [GitHub actions](https://docs.github.com/en/actions) that run on every pull request. For instance, org-wide [self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) will give us better test result UI, test coverage across more platforms (including more TPUs and GPUs), extra security features, and better availability (e.g. runners spread geographically).

We are moving the repository to take advantage of these and any other nice things going forward. We see a bonus opportunity to have the JAX core listed with its closely related extensions in one place. This move does not affect anything else about how JAX is developed. Expect the only change for now to be the URL and improved CI.

**Comment by 8bitmp3** [comment]
![](https://media1.tenor.com/m/j2vJg6CK4MgAAAAC/im-ready-cat.gif)
